### DISCUSSION TITLE
Batch processing of individual Fourier distributions

### DISCUSSION THREAD

**Discussion by Jorge-LF** [tensionhead/pyBOAT#20] [Ideas]
Hi,

I would like to suggest a potential implementation.
Right now we can extract the data (CSV file) of the Fourier distribution of a specific signal by Analyze Signal > Estimate Fourier > Save results. This is very convenient for extracting the period with the higher power across the whole time series.
I would like to be able to do this in batch.
In the Analyze All tab I can only export the global fourier estimate which averages the fourier distributions of all the data. Most times I want to obtain a single CSV file with the Fourier distribution for each timeseries data (similar to what happens when batch processing the ridge readouts).

I would be very grateful if something like this could be implemented so I don't have to save the single Fourier distributions one by one.

Thanks a lot for maintaining this amazing tool updated!

**Comment by tensionhead** [comment]
Hey @Jorge-LF,

yes that's a good idea.. originally I thought the individual Fourier estimates are not that valuable, as in principle they're just a fancy (time averaging Wavelet spectra) way to obtain a (pretty smoothed) FFT result. But adding this to the batch outputs should be no big deal! You want to open an issue yourself about it, 1 sentence and maybe a link to this thread would be enough :slightly_smiling_face: 

EDIT: Oh I see the github web UI now allows to transform this into an issue with 1 click, ok I'll do it

**Comment by tensionhead** [comment]
Hey @Jorge-LF the feature got implemented with the just released [pyBOAT 0.9.12](https://github.com/tensionhead/pyBOAT/releases/tag/v0.9.12):

![ExportFourierEstimates](https://github.com/tensionhead/pyBOAT/assets/10595420/bee8d377-3ddc-4ca2-bc06-892ac939687e)

It's already on pip, conda version should follow in a few hours.. lemme know when you had time to check it out if it does what it's supposed to do!

**Comment by Jorge-LF** [comment]
Thank you for the update.
Works great and it is exactly what I needed for my analysis of large number of individual signals.
